### PR TITLE
Tag a new release on each PR to main

### PR DESCRIPTION
**Pull request by cppvik**
The `.version` file in the repo root should contain the new version. If such a tag already exists, it won't be updated.

Should be reviewed & merged along with https://github.com/scribe-security/valint/pull/201.

### CODE REVIEW

File: .github/workflows/sync-valint-version.yaml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Read version from file
+        id: read_version
+        run: echo "version=$(cat .version)" >> $GITHUB_OUTPUT
```
Comment (houdini91) [inline .github/workflows/sync-valint-version.yaml:24]: Whats this file?

File: .github/workflows/sync-valint-version.yaml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        id: check_tag
+        run: |
+          tag_name=$(echo "${{ steps.read_version.outputs.version }}")
+          git fetch --tags
+          if git rev-parse "$tag_name" >/dev/null 2>&1; then
+            echo "Tag $tag_name already exists"
+            echo "::set-output name=tag_exists::true"
+          else
+            echo "Tag $tag_name does not exist"
+            echo "::set-output name=tag_exists::false"
```
Comment (houdini91) [inline .github/workflows/sync-valint-version.yaml:36]: this is no longer how github support set output. 
https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#example-masking-a-generated-output-within-a-single-job

File: .github/workflows/sync-valint-version.yaml (hunk 1 above)
Comment (cppvik) [reply .github/workflows/sync-valint-version.yaml:24]: It's pushed by the valint pipeline. I'll add a dummy one.

File: .github/workflows/sync-valint-version.yaml [hunk 3]
```diff
@@ -0,0 +1,44 @@
+name: Sync Valint Version
+
+on:
+  pull_request:
```
[the remark this answers is not in the document]
Comment (cppvik) [reply .github/workflows/sync-valint-version.yaml:4]: I have a check on that later:

```yaml
if: github.event.pull_request.merged == true && ...
```

File: .github/workflows/sync-valint-version.yaml (hunk 2 above)
Comment (cppvik) [reply .github/workflows/sync-valint-version.yaml]: fixed in 4e810d184318d284c86304025bd1e74334316680

**Review by houdini91 — APPROVED** [review]

**Pull request merged**